src/classes/STG_PanelOppNaming_CTRL.cls

```diff
@@ -0,0 +1,222 @@
+    */
+    public override string idPanel() { return 'idPanelOppNaming'; }
+
+    /*********************************************************************************************************
+    * @description Whether we are running the batch opportunity naming process 
+    */
+    public boolean isRunningBatch { get; set; }
+
+    /*********************************************************************************************************
+    * @description Action Method to delete a specific settings instance.
+    * @return null 
+    */
+    public PageReference delONS() {
+        delete mapOppNameSettings.get(currentONSName);
+        mapOppNameSettings = Opportunity_Naming_Settings__c.getAll();
+        return null;
+    }
+
+    /*********************************************************************************************************
+    * @description Action Method to enter Edit mode for a specific setting instance.
+    * @return null 
+    */
+    public PageReference editONS() {
+        currentONS = mapOppNameSettings.get(currentONSName);
+        super.editSettings();
+        return null;
+    }
+
+    /*********************************************************************************************************
+    * @description Action Method to create a new settings object in memory, then enter Edit mode.
+    * @return null 
+    */
+    public PageReference newONS() {
+        currentONS = new Opportunity_Naming_Settings__c(
+            Name = System.now().format()
+        );
+        super.editSettings();
+        return null;
+    }
+
+    /*********************************************************************************************************
+    * @description Action Method to save the current edited setting, refresh the list of settings, and return
+    * to the main page.
+    * @return null 
+    */
+    public PageReference saveONS() {
+        upsert currentONS;
+        mapOppNameSettings = Opportunity_Naming_Settings__c.getAll();
+        super.cancelEdit();
+        return null;
+    }
+
+    /*********************************************************************************************************
+    * @description Action Method to run batch opportunity renaming.
+    * @return null 
+    */
+    public PageReference runBatch() {
+        isRunningBatch = true;
+        OPP_OpportunityNaming_BATCH batch = new OPP_OpportunityNaming_BATCH();
+        id batchProcessId = database.executeBatch(batch, 200);
+        return null;
+    }
+    
+    /*********************************************************************************************************
+    * @description Returns wwhether the custom name formats are valid or not, and adds a Page Message for
+    * any errors.
+    * @return boolean 
+    */
+    public boolean isValidNameFormats() {
+        // make sure any custom name formats are valid.
+        if (STG_Panel.stgService.stgHH.npo02__Advanced_Household_Naming__c) {
```

Comment:
I didn't get a chance to implement checking valid name formats. I have a story to do it this sprint. Nothing calls this method currently.